src/Sarif.Viewer.VisualStudio.Core/Tags/SarifLocationErrorTagger.cs

```diff
@@ -95,14 +98,28 @@ public IEnumerable<ITagSpan<IErrorTag>> GetTags(NormalizedSnapshotSpanCollection
 
             foreach (SnapshotSpan span in spans)
             {
-                foreach (ISarifLocationTag possibleTag in this.currentTags.Where(currentTag => currentTag.PersistentSpan.Span != null))
+                Dictionary<(int start, int end), (List<IErrorTag> tagList, SnapshotSpan snapshotSpan)> groupedBySpan = new Dictionary<(int start, int end), (List<IErrorTag> tagList, SnapshotSpan snapshotSpan)>();
```

Comment:
>Dictionary<(int start, int end), (List<IErrorTag> tagList, SnapshotSpan snapshotSpan)> [](http://example.com/codeflow?start=16&length=86)

var
(I have never liked var, but this is a particularly good place for it to improve readability.)